scripts/skills/actives/rf_hook_shield_skill.nut

```diff
@@ -110,20 +112,38 @@ this.rf_hook_shield_skill <- ::inherit("scripts/skills/skill", {
 		if (targetEntity.isAlliedWith(this.getContainer().getActor()))
 			return false;
 
-		return targetEntity.isArmedWithShield() && this.skill.onVerifyTarget(_originTile, _targetTile) && !targetEntity.getSkills().hasSkill("effects.shieldwall") && !targetEntity.getSkills().hasSkill("effects.rf_hooked_shield");
+		return targetEntity.isArmedWithShield() && this.skill.onVerifyTarget(_originTile, _targetTile) && !targetEntity.getSkills().hasSkill("effects.rf_hooked_shield");
 	}
 
 	function onUse( _user, _targetTile )
 	{
 		local targetEntity = _targetTile.getEntity();
+
+		if (!this.m.IsSpent && targetEntity.getID() == this.m.LastTargetID)
+		{
+			_user.setActionPoints(::Math.min(_user.getActionPointsMax(), this.getActionPointCost()));
```

Comment:
`getActionPointCost` does not take into account the vanilla property `.getCurrentProperties().AdditionalActionPointCost`, see `function use` from the `skill.nut`

Reply:
I submitted a [bug report](https://battlebrothersgame.com/forums/topic/problems-with-getactionpointcost-and-similar-functions-in-skill-nut/) for that several years ago but Overhype took no action against it. So I have no idea whether that is intended behavior or not. In any case `getActionPointCost` is what we are supposed to use to get the AP cost of a skill - we don't really have a better choice. Unless of course we decide to change this vanilla behavior in Modular Vanilla as per my suggestion in the bug report.

Reply:
An alternative approach we can use right now is use `onBeforeAnySkillExecuted` and `onAnySkillExecuted` to check for the AP cost of the skill.